### PR TITLE
feat: Enable Rule CA1805

### PR DESCRIPTION
**Pull request by workgroupengineering**
## What does the pull request do?

Address [CA1805](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1805): Do not initialize unnecessarily

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues

Part of #8944

### CODE REVIEW

**Comment by robloo** [comment]
I'm not really a fan with what this does to code readability in some cases. Honestly, this type of thing should be taken care of in the compiler anyway as it seems an easy optimization.

I also don't think there is any practical performance impact here and these types of chances are unnecessary. I would be careful before doing too many changes for code analyzers...

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024596-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Base/Input/PointerPoint.cs [hunk 1]
```diff
@@ -33,55 +33,55 @@ public PointerPoint(IPointer pointer, Point position, PointerPointProperties pro
     /// <summary>
     /// Provides extended properties for a PointerPoint object.
     /// </summary>
-    public struct PointerPointProperties
+    public record PointerPointProperties
```
Comment (maxkatz6) [inline src/Avalonia.Base/Input/PointerPoint.cs:36]: It changes "struct" type to a "class" type. I don't think it's intended and probably shouldn't be part of this PR.

File: src/Windows/Avalonia.Win32/Input/Imm32InputMethod.cs [hunk 2]
```diff
@@ -17,7 +17,7 @@ class Imm32InputMethod : ITextInputMethodImpl
         private WindowImpl _parent;
         private ITextInputMethodClient _client;
 
-        private Imm32CaretManager _caretManager = new();
+        private Imm32CaretManager _caretManager;
```
Comment (maxkatz6) [inline src/Windows/Avalonia.Win32/Input/Imm32InputMethod.cs:20]: Structs are special here.
`default(Imm32CaretManager)` is not the same as `new Imm32CaretManager()`.
In first case ctor won't be called, but in second case it will.
I know, that Imm32CaretManager doesn't have any public ctors, but I am not sure about removing initialization anyway. Ctor might be added in the future.

**Comment by maxkatz6** [comment]
I agree that this analyzer isn't really useful in terms of performance. You can see that removing default value doesn't really change anything in the [asm output](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunGBLAHYYGAQQYBeBgAYA3DS496TckgaDhAWQAUASg4KeigOwMBMAO6tdAOhFzqhgL41HQA).

But it rather a code style choice.
